File: q90-q99/q94/q94.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        TreeNode* cur = root;
        vector<int> inorder_nodes;
        stack<TreeNode*> node_stack;
        while (cur != NULL || !node_stack.empty()) {
            while (cur != NULL) {
                node_stack.push(cur);
                cur = cur->left;
            }
            cur = node_stack.top();
            node_stack.pop();
            inorder_nodes.push_back(cur->val);
            cur = cur->right;
        }
        return inorder_nodes;
    }
};

File: q90-q99/q94/q94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        inorder_nodes = []
        node_stack = []
        cur = root
        while cur != None or len(node_stack) != 0:
            while cur != None:
                node_stack.append(cur)
                cur = cur.left
            cur = node_stack.pop()
            inorder_nodes.append(cur.val)
            cur = cur.right
        return inorder_nodes
